client/app/pages/queries/QuerySource.jsx

```diff
@@ -44,60 +51,83 @@ function QuerySource(props) {
   const [query, setQuery] = useState(props.query);
   const [dataSources, setDataSources] = useState([]);
   const [dataSourcesLoaded, setDataSourcesLoaded] = useState(false);
-  const dataSource = useMemo(() => (find(dataSources, { id: query.data_source_id }) || null), [query, dataSources]);
+  const dataSource = useMemo(
+    () => find(dataSources, { id: query.data_source_id }) || null,
+    [query, dataSources]
+  );
   const [schema, setSchema] = useState([]);
   const refreshSchemaTokenRef = useRef(null);
 
   useEffect(() => {
-    recordEvent('view_source', 'query', query.id);
+    recordEvent("view_source", "query", query.id);
 
     let isCancelled = false;
-    DataSource.query().$promise.then((data) => {
+    DataSource.query().$promise.then(data => {
       if (!isCancelled) {
         setDataSources(data);
         setDataSourcesLoaded(true);
       }
     });
 
-    return () => { isCancelled = true; };
+    return () => {
+      isCancelled = true;
+    };
   }, []);
 
-  const reloadSchema = useCallback((refresh = undefined) => {
-    const refreshToken = Math.random().toString(36).substr(2);
-    refreshSchemaTokenRef.current = refreshToken;
-    getSchema(dataSource, refresh).then((data) => {
-      if (refreshSchemaTokenRef.current === refreshToken) {
-        setSchema(data);
-      }
-    });
-  }, [dataSource]);
+  const reloadSchema = useCallback(
+    (refresh = undefined) => {
+      const refreshToken = Math.random()
+        .toString(36)
+        .substr(2);
+      refreshSchemaTokenRef.current = refreshToken;
+      getSchema(dataSource, refresh).then(data => {
+        if (refreshSchemaTokenRef.current === refreshToken) {
+          setSchema(data);
+        }
+      });
+    },
+    [dataSource]
+  );
 
   useEffect(() => {
     reloadSchema();
   }, [reloadSchema]);
 
-  useEffect(() => (() => {
-    // cancel pending operations
-    refreshSchemaTokenRef.current = null;
-  }), []);
-
-  const handleDataSourceChange = useCallback((dataSourceId) => {
-    localStorage.lastSelectedDataSourceId = dataSourceId;
-    if (query.data_source_id !== dataSourceId) {
-      const newQuery = clone(query);
-      newQuery.data_source_id = dataSourceId;
-      setQuery(newQuery);
-    }
-  }, [query]);
+  useEffect(
+    () => () => {
+      // cancel pending operations
+      refreshSchemaTokenRef.current = null;
+    },
+    []
+  );
+
+  const handleDataSourceChange = useCallback(
+    dataSourceId => {
```

Comment:
here ^ (and few similar places) - braces were required when arrow function has body (?)